src/iop/colorequal.c

```diff
@@ -314,8 +309,32 @@ static inline float _deg_to_rad(const float angle)
   return (angle + ANGLE_SHIFT) * M_PI_F / 180.f;
 }
 
+/* We use precalculated data for the logistic weighing function for performance and stability
+   and do linear interpolation at runtime. Avoids banding effects and allows a sharp transition.
+   By uncommenting SATCURVE the function will be shown in the masking modes.
+*/
+#define SATCURVE
+static float satweights[SATSIZE + 1];
+static void _init_satweights(void)
+{
+  for(int i = 0; i < SATSIZE + 1; i++)
+  {
+    const double val = 1.0 / (double)SATSIZE * (double)i;
+    const double coef = exp(-(60.0 * (val - 0.1)));
+    satweights[i] = (float)(fmax(1.0 / (1.0 + coef), 0.0));
+  }
+}
+
+static inline float _get_satweight(const float sat)
+{
+  const float isat = (float)SATSIZE * CLIP(sat);
+  const float base = floorf(isat);
+  const int i = (int)base;
```

Comment:
For non-negative values (which we have here thanks to the CLIP), `(int)x` is always equal to `(int)floorf(x)`.  For negative values, we'd get different results, because casting to int truncates toward zero.

Other linear interpolation tables in the dt codebase are all as far as I can recall 64K entries.  Probably worth a quick check to see how much of a difference that makes at the most extreme settings of the module's sliders.

Reply:
Yes about the int cast, just a missed residule while working..

About the table size, I initially used just 1024 and stepped up to 4k for safety but even then there were absolutely no diffs.